### ISSUE TITLE
🚀 Feature: Short direct file links

### ISSUE THREAD

**Issue by mariuszste** [stonith404/pingvin-share#304]
### 🔖 Feature description

The `/s/<id>` urls are nice but they aren't direct links to file. It would be really useful if it was possible to reference access single files in a similar manner. Maybe something like `/f/<short id>/[optional file name]`. The current direct links are very long and use UUIDv4 which is not the most friendly thing to type on a keyboard.

### 🎤 Pitch

I often need to transfer single files between virtual machines and often these virtual machines don't have an easy way to copy and paste a long url. They also often lack a user interface or only have access to very old browsers. Having a short, self-hosted, direct download url that auto expires would be supper useful.

Some tools (like wget) will by default ignore the `Content-Disposition` header meaning that the name of the downloaded file will just be whatever was after the last `/`. Having the ability for a user to put anything there would make it trivial to download the file with the desired file name without needing an additional rename step. Some applications also make assumptions about the file type based on the file extension they see in the url rather than the `Content-Type` header. 

The file name should of course be optional to allow for nice looking, easy to type short urls.

**Comment by W1BTR** [comment]
Would also love this! :)

**Comment by Xaero252** [comment]
I agree with the above, though I'm not sure how you would implement it easily. My thought would be to shorten with: https://your.domain/s/<share-name>/<file-index>[/file name]. For a multi-file share you would then have access to every file shared via it's index, with the ability to optionally specify a file name for the aforementioned `Content-Disposition` header issue. You could also add `z` as the index to get the zip for a multi-file share.
